src/index.c

```diff
@@ -1337,57 +1331,57 @@ static int index_insert(
 	bool trust_mode,
 	bool trust_id)
 {
-	int error = 0;
-	size_t path_length, position;
 	git_index_entry *existing, *best, *entry;
+	size_t position;
+	int error;
 
 	assert(index && entry_ptr);
 
 	entry = *entry_ptr;
 
-	/* make sure that the path length flag is correct */
-	path_length = ((struct entry_internal *)entry)->pathlen;
-	index_entry_adjust_namemask(entry, path_length);
+	/* Make sure that the path length flag is correct */
+	index_entry_adjust_namemask(entry, ((struct entry_internal *)entry)->pathlen);
```

Comment:
This is personal preference, but the old version looks clearer to me as we're not doing the casting and dereferencing in the function call.

Reply:
Okay, I've dropped this commit. I prefer not having the variable, but taste is different and I don't care that much